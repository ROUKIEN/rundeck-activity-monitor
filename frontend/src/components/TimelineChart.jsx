import React from 'react'
import { Chart } from 'react-google-charts'
import { parseExecutions } from '../utils'

const Loader = () =>
  <div className="message">
    <div className="message-body">Loading, please wait...</div>
  </div>

const FailedToLoad = () =>
  <div className="message">
    <div className="message-body">Failed to render executions.</div>
  </div>

function TimelineChart({ executions }) {
  const { data } = parseExecutions(executions)

  return (
    <Chart
      chartType="Timeline"
      data={data}
      options={
        {
          timeline: {
            showBarLabels: false
          }
        }
      }
      width="100%"
      height="600px"
      loader={<Loader></Loader>}
      errorElement={<FailedToLoad></FailedToLoad>}
      chartEvents={
        [
          {
            eventName: 'select',
            callback: (e) => {
              const selection = e.chartWrapper.getChart().getSelection()
              const execution = executions[selection[0].row]
              window.open(execution.execution_permalink, '_blank').focus()
            }
          }
        ]
      }
    />
  )
}

export default TimelineChart
