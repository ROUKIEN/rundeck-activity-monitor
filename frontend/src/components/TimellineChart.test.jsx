import React from 'react'
import renderer from 'react-test-renderer'
import { expect, test } from 'vitest'
import TimelineChart from './TimelineChart'

test('TimelineChart renders correctly', () => {
  const data = []
  const tc = renderer.create(
    <TimelineChart data={data}></TimelineChart>
  )
  expect(tc).toBeDefined()
})
