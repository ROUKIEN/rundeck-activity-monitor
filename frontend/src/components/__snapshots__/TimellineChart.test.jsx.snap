// Vitest Snapshot v1

exports[`TimelineChart renders correctly > Rendered list without any facet selected 1`] = `
<div
  className="message"
>
  <div
    className="message-body"
  >
    Loading, please wait...
  </div>
</div>
`;
